### DISCUSSION TITLE
Handler never called when using throttle with key events

### DISCUSSION THREAD

**Discussion by sterat** [alpinejs/alpine#1895] [7. Bugs (archived)]
If you add a handler for keyup or keydown events combined with throttle, then the handler is never called.

```html
<div x-data>
<input type="text" x-on:keyup.throttle.1000ms="console.log($event)">
</div>
```

This is because in `isListeningForASpecificKeyThatHasntBeenPressed()` it filters out a list of known event types and assumes that any remaining modifiers are keys to wait for.  Unfortunately it omits throttle from the list of known events, so it waits for an invalid key to be pressed.

**Comment by KevinBatdorf** [comment]
For reference:

https://github.com/alpinejs/alpine/blob/b46c41fa240cd8af2dcaa29fb60fb1db0389c95a/packages/alpinejs/src/utils/on.js#L123-L162

Is this something you've solved already and could submit a PR for?

**Reply by sterat** [reply]
I do have a fix.  It involves a large change but I will do a PR.

**Reply by KevinBatdorf** [reply]
Cool, nice one. (It's here for reference: https://github.com/alpinejs/alpine/pull/1911)

You may want to clean up the code style a bit to match the repo. Few things that stand out to me are:

- Caleb tends to return early, so in `timeArg` I would flip the logic to return early instead. The other if/else block is ok though, I think (because nothing is returning).
- Some lines have semicolons there. A pre-commit linter would be ideal here actually, but that's for another PR I suppose.
- Is it possible to do it without changing the `modifiers` data type? I believe it's an array everywhere else in the codebase, so making it an object here would be confusing, I think. Could you just make the `keyModifiers` a second array or something? This might be my personal preference though, but I try to have more things be the same so your brain doesn't have to second guess anything. 

Also, those are just my thoughts. I'm not the maintainer, but a user like yourself (i just comment a lot)! So take it with a grain of salt, of course.

**Comment by sterat** [comment]
Thanks I made a few changes based on your comments.
